src/main/scala/beam/agentsim/agents/ridehail/RideHailAgent.scala

```diff
@@ -176,15 +194,63 @@ class RideHailAgent(
     case Event(TriggerWithId(InitializeTrigger(tick), triggerId), data) =>
       beamVehicles.put(vehicle.id, ActualVehicle(vehicle))
       vehicle.becomeDriver(self)
+      vehicle.manager = Some(rideHailManager)
       eventsManager.processEvent(
         new PersonDepartureEvent(tick, Id.createPersonId(id), Id.createLinkId(""), "be_a_tnc_driver")
       )
       eventsManager.processEvent(new PersonEntersVehicleEvent(tick, Id.createPersonId(id), vehicle.id))
-      goto(Idle) replying CompletionNotice(triggerId) using data
-        .copy(currentVehicle = Vector(vehicle.id))
+      val isTimeForShift = shifts.isEmpty || shifts.get
+        .find(shift => shift.lowerBound <= tick && shift.upperBound >= tick)
+        .isDefined
+      if (isTimeForShift) {
+        rideHailManager ! NotifyVehicleIdle(
+          vehicle.id,
+          vehicle.spaceTime,
+          PassengerSchedule(),
+          vehicle.getState,
+          Some(triggerId)
+        )
+        holdTickAndTriggerId(tick, triggerId)
```

Comment:
I was looking at this trait and it might need redefined (`updateTickAndTriggerId` or something). This is not a collection, but instead more of an overwrite to be the latest at all times. If you knew that and overwriting upon another run is fine, then this is good to go.